Tasks/AzureKeyVaultV1/operations/azure-arm-keyvault.ts

```diff
@@ -26,6 +26,44 @@ export class KeyVaultClient extends azureServiceClient.ServiceClient {
         this.apiVersion = '2016-10-01';
     }
 
+    public async invokeRequest(request: webClient.WebRequest): Promise<webClient.WebResponse> {
+        try {
+            var response = await this.beginRequest(request);
+            if (response.statusCode == 401) {
+                var vaultResourceId = this.getValidVaultResourceId(response);
+                if(!!vaultResourceId) {
+                    console.log(tl.loc("RetryingWithVaultResourceIdFromResponse", vaultResourceId));
+                    
+                    this.getCredentials().activeDirectoryResourceId = vaultResourceId; // update vault resource Id
+                    this.getCredentials().getToken(true); // Refresh authorization token in cache
+                    var response = await this.beginRequest(request);
+                }
+            }
+            return response;
+        } catch(exception) {
+            throw exception;
+        }
+    }
+
+    public getValidVaultResourceId(response: webClient.WebResponse) {
```

Comment:
Can we have some test around this. I am worried if the response changes, we might be broken.

Reply:
As discussed offline, since the version of the API will remain the same assuming response will be in the same format. 

Currently, there is no L0 for the key vault task. Opening bug for that.